web/massastation/src/layouts/LayoutStation/LayoutStation.tsx

```diff
@@ -22,6 +22,18 @@ interface NetworkRequest {
 export function LayoutStation({ ...props }) {
   const { children, navigator, onSetTheme, storedTheme } = props;
 
+  const {
+    data: massaStationVersion,
+    isSuccess: getMassaStationVersionSuccess,
+  } = useResource<string>('version');
+
+  const [currentStationVersion, setCurrentStationVersion] =
+    useState<string>('');
+
+  useEffect(() => {
+    if (massaStationVersion) setCurrentStationVersion(massaStationVersion);
+  }, [getMassaStationVersionSuccess, massaStationVersion]);
```

Comment:
Why do we need a **useEffect** if we are hitting the endpoint!? 🤔

Reply:
i belive the useressource is async

Reply:
but in case of a slow internet connection are we not risking that the version number is never initialized before the screen is rendered ?

Reply:
Loading state...